service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -415,16 +414,18 @@ private void searchAttributesWithScopes(final E entity, final Vertex vertex, fin
                             Attribute.class);
                     if (deserializedAttributes != null) {
                         attributes.addAll(deserializedAttributes);
-                        checkTraversalLimitOrFail(attributes);
+                        checkTraversalLimitOrFail(entity, attributes);
                     }
                 });
         entity.setAttributes(attributes);
         entity.setAttributesAsJson(JSON_UTILS.serialize(attributes));
     }
 
-    private void checkTraversalLimitOrFail(final Set<Attribute> attributes) {
+    private void checkTraversalLimitOrFail(final E e, final Set<Attribute> attributes) {
         if (attributes.size() > this.traversalLimit) {
```

Comment:
I don't fully comprehend the code at this point but at first glance this seems to be missing the point. The whole point of the traversal limit is to limit the traversal. Why would we ever allow the attribute size to exceed this?